librz/core/cmd/cmd_regs.c

```diff
@@ -0,0 +1,787 @@
+				pj_ks(pj, "name", r->name);
+				pj_kn(pj, "size", r->size);
+				pj_kn(pj, "offset", r->offset);
+				pj_end(pj);
+			}
+		}
+		pj_end(pj);
+		pj_end(pj);
+		break;
+	}
+	default:
+		break;
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_reg_profile_comments_handler(RzCore *core, RzReg *reg, int argc, const char **argv) {
+	if (reg->reg_profile_cmt) {
+		rz_cons_println(reg->reg_profile_cmt);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_reg_profile_open_handler(RzCore *core, RzReg *reg, int argc, const char **argv) {
+	rz_return_val_if_fail(argc > 1, RZ_CMD_STATUS_WRONG_ARGS);
+	rz_reg_set_profile(reg, argv[1]);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_reg_cond_handler(RzCore *core, RzReg *reg, RzCmdRegSync sync_cb, int argc, const char **argv) {
+	SYNC_READ(RZ_REG_TYPE_ANY);
+	RzRegFlags *rf = rz_reg_cond_retrieve(reg, NULL);
+	if (!rf) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_cons_printf("| s:%d z:%d c:%d o:%d p:%d\n",
+		rf->s, rf->z, rf->c, rf->o, rf->p);
+	for (int i = 0; i < RZ_REG_COND_LAST; i++) {
+		rz_cons_printf("%d %s\n",
+			rz_reg_cond_bits(reg, i, rf),
+			rz_reg_cond_to_string(i));
+	}
+	free(rf);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_reg_cc_handler(RzCore *core, RzReg *reg, int argc, const char **argv) {
+	char *s = rz_reg_profile_to_cc(reg);
+	if (s) {
+		rz_cons_printf("%s\n", s);
+		free(s);
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_regs_diff_handler(RzCore *core, RzReg *reg, RzCmdRegSync sync_cb, int argc, const char **argv) {
+	SYNC_READ(RZ_REG_TYPE_ANY);
+	RzListIter *iter;
+	RzRegItem *item;
+	rz_list_foreach (reg->allregs, iter, item) {
+		ut64 newval = rz_reg_get_value(reg, item);
+		rz_reg_arena_swap(reg, false);
+		ut64 oldval = rz_reg_get_value(reg, item);
+		rz_reg_arena_swap(reg, false);
+		ut64 delta = newval - oldval;
+		if (delta) {
+			rz_cons_printf(
+				"%s = 0x%" PFMT64x " was 0x%" PFMT64x " delta 0x%" PFMT64x "\n",
+				item->name, newval, oldval, delta);
+		}
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_regs_prev_handler(RzCore *core, RzReg *reg, int argc, const char **argv, RzCmdStateOutput *state) {
+	const char *filter = argc > 1 ? argv[1] : NULL;
+	rz_reg_arena_swap(reg, false);
+	RzCmdStatus r = show_regs_handler(core, reg, NULL, filter, state); // sync_cb = NULL on purpose to not overwrite
+	rz_reg_arena_swap(reg, false);
+	return r;
+}
+
+RZ_IPI RzCmdStatus rz_regs_fpu_handler(RzCore *core, RzReg *reg, RzCmdRegSync sync_cb, int argc, const char **argv) {
+	// TODO: everything here needs to be rewritten. It was taken from the old "drf" command.
+	if (argc <= 1) {
+		// TODO: Do not use this hack to print fpu register:
+		// By sending a negative value, this is signalling all the way through to the debug plugin,
+		// which then does the printing.
+		// This should be rewritten directly above the RzReg.
+		SYNC_READ(-RZ_REG_TYPE_FPU);
+		return RZ_CMD_STATUS_OK;
+	}
+	char *name = rz_str_trim_dup(argv[1]);
```

Comment:
Maybe you could just copy the logic for the newshell parsing from the `e` command handler?

Reply:
I did check `e` but the logic is a bit different there. The arg here has the `_LAST` flag and supports one assignment with spaces.